OptimizelySDKCore/OptimizelySDKCore/Optimizely.m

```diff
@@ -422,6 +431,9 @@ - (nullable NSString *)variableString:(nonnull NSString *)variableKey
         } else {
             [self.errorHandler handleError:variableUnknownForVariableKey];
         }
+        if (callback) {
+            callback(*error);
+        }
         
```

Comment:
We decided to have an error return for variableString (and all the live variable methods) so that the user knows when there is an invalid value for a live variable value (you can't return a nil for a primitive value). Activate does not need to return an error. The user calls activate and if the event dispatch fails, we save that event and retry later.